### PR TITLE
Add set_st custom macropad

### PR DESCRIPTION
**Pull request by set-st**
## Description

Custom macropad based on rp2040

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by keyboard-magpie** [comment]
Invalid PR as you have deleted the PR template which is there for good reason, as well as extra files included.

**Comment by set-st** [comment]
fixed

**Comment by zvecr** [comment]
> fixed

Nope, you still have many issues covered by the PR checklist. This includes:
* VIA json which need to be removed
* vscode settings file which needs to be removed
* Content which should be migrated to info.json

Also, you should not close/open new PRs with the same content.

**Comment by tzarc** [comment]
And, one PR per board.

**Comment by set-st** [comment]
What is wrong with encoder_map? Its compiling fine locally.

Run QMK_CHANGES=$(echo -e 'keyboards/handwired/set_st/rp2040/config.h keyboards/handwired/set_st/rp2040/halconf.h keyboards/handwired/set_st/rp2040/info.json keyboards/handwired/set_st/rp2040/keymaps/default/keymap.c keyboards/handwired/set_st/rp2040/keymaps/default/rules.mk keyboards/handwired/set_st/rp2040/keymaps/via/keymap.c keyboards/handwired/set_st/rp2040/keymaps/via/rules.mk keyboards/handwired/set_st/rp2040/mcuconf.h keyboards/handwired/set_st/rp2040/readme.md keyboards/handwired/set_st/rp2040/rp2040.c keyboards/handwired/set_st/rp2040/rules.mk' | sed 's/ /\n/g')
linting handwired/set_st/rp2040
☒ handwired/set_st/rp2040: Invalid keyboard level feature detected - encoder_map
☒ Lint check failed for: handwired/set_st/rp2040
Error: Process completed with exit code 1.

**Comment by fauxpark** [comment]
Encoder map is not allowed at keyboard level. It will not compile in the Configurator, because it generates its own keymap which will not have an encoder_map array.

File: keyboards/handwired/set_st/rp2040/config.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+#ifdef RGB_MATRIX_ENABLE
+
+     /* RGB Defines */
+#    define RGB_MATRIX_LED_COUNT 10
+
+     /* Enable Framebuffer and keypress effects */
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#    define RGB_MATRIX_KEYPRESSES
+
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/config.h:65]:
```suggestion
```

File: keyboards/handwired/set_st/rp2040/config.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#    define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#    define ENABLE_RGB_MATRIX_SPLASH
+#    define ENABLE_RGB_MATRIX_MULTISPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#    define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#endif
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/config.h:109]:
```suggestion
```
The RGB Matrix effects config above should be moved to info.json.

File: keyboards/handwired/set_st/rp2040/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef VIA_ENABLE
+  #include "via.h"
+#endif
+
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/keymaps/default/keymap.c:22]:
```suggestion
```
`default` keymaps must not have VIA enabled.

File: keyboards/handwired/set_st/rp2040/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef VIA_ENABLE
+  #include "via.h"
+#endif
+
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/keymaps/via/keymap.c:22]:
```suggestion
```
This serves no purpose anyway as via.h is already included through `QMK_KEYBOARD_H`.

File: keyboards/handwired/set_st/rp2040/rp2040.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    { NA,  NA, NA, NA, NA },
+    {  0, 1, 2, 3, 4 },
+    {  9, 8, 7, 6, 5 },
+}, {
+    { 0,  0 }, { 112,  0 }, { 224,  0}, { 0, 21 }, { 112, 21 },
+    { 224, 21}, { 0, 42 }, { 112, 42 }, { 224, 42}, { 0, 64 },
+}, {
+    4, 4, 4, 4, 4,
+    4, 4, 4, 4, 4,
+} };
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/rp2040.c:34]: LED config should be moved to info.json and this file deleted.

File: keyboards/handwired/set_st/rp2040/rules.mk [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Build Options
+#   change yes to no to disable
+#
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
+RGB_MATRIX_ENABLE = yes
+OLED_ENABLE = yes
+OLED_DRIVER = ssd1306
+OLED_TRANSPORT = i2c
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/rules.mk:10]:
```suggestion
# This file intentionally left blank
```

File: keyboards/handwired/set_st/rp2040/info.json [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    "manufacturer": "PDAINUA",
+    "url": "https://pda.in.ua",
+    "maintainer": "set-st",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
```
Comment (fauxpark) [inline keyboards/handwired/set_st/rp2040/info.json:12]:
```suggestion
        "nkro": true,
        "encoder": true,
        "rgb_matrix": true,
        "oled": true
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**